flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -169,6 +173,26 @@ public class CheckpointingOptions {
                                     + "deactivated. Local recovery currently only covers keyed state backends "
                                     + "(including both the EmbeddedRocksDBStateBackend and the HashMapStateBackend).");
 
+    /**
+     * This option configures local backup for the state backend, which indicates whether to make
+     * backup checkpoint on local disk. If not configured, fallback to {@link
+     * StateRecoveryOptions#LOCAL_RECOVERY}. By default, local backup is deactivated. Local backup
+     * currently only covers keyed state backends (including both the EmbeddedRocksDBStateBackend
+     * and the HashMapStateBackend).
+     */
+    public static final ConfigOption<Boolean> LOCAL_BACKUP_ENABLED =
+            ConfigOptions.key("execution.checkpointing.local-backup.enabled")
+                    .booleanType()
+                    .defaultValue(StateRecoveryOptions.LOCAL_RECOVERY.defaultValue())
+                    .withFallbackKeys(StateRecoveryOptions.LOCAL_RECOVERY.key())
```

Comment:
I'm thinking does it make sense to make `'LOCAL_BACKUP_ENABLED'` as fallback keys of the new `'LOCAL_RECOVERY'` instead of the reverse order? Since whether to recover from local is typically set aligned with whether to make local backup.
And both options should have same deprecated keys 'state.backend.local-recovery'.

Reply:
1. First, I think we should support enabling both localBackup and localRecovery with only one parameter configured, 
which aligns with the previous behavior.

2. IMO, using the StateRecoveryOptions.LOCAL_RECOVERY as this single parameter is more suitable, 
as its semantic is closer to the legacy parameter, thereby simplifying comprehension and migration for users.

Based on 1 and 2, I think StateRecoveryOptions.LOCAL_RECOVERY should act as the fallback key for CheckpointingOptions.LOCAL_BACKUP_ENABLED?
That is, if user only configure StateRecoveryOptions.LOCAL_RECOVERY, both localRecovery and localBackup can be enabled.

Reply:
OK I get it. +1 for current fallback strategy. But why there is only `.withDeprecatedKeys(CheckpointingOptions.LOCAL_RECOVERY.key())` for LOCAL_RECOVERY and not for LOCAL_BACKUP_ENABLED

Reply:
It makes sense that both options have same deprecated key 'state.backend.local-recovery'. I have refined it!